src/tools/pub/cmd/pub-outdated.md

```diff
@@ -0,0 +1,251 @@
+
+{% prettify nocode tag=pre+code %}
+$ pub outdated [options]
+{% endprettify %}
+
+Use `pub outdated` to identify out-of-date [package dependencies][]
+and get advice on how to update them.
+<span style="color:red">[PENDING: Should we motivate why you want to do this?
+Null safety might be the biggest near-term motivation,
+but are there other major reasons
+that make it worth the effort to update and test your code?]</span>
+
+[package dependencies]: /tools/pub/dependencies
+
+Here's an overview of how you can use `pub outdated`
+to help you update package dependencies:
+
+1. In the top directory of your app or package,
+   **run `pub upgrade`**
+   to create or update the `pubspec.lock` file.
+2. **Run `pub outdated`** and inspect its output.
+3. If the output reports out-of-date dependencies,
+   **update the version dependencies** in `pubspec.yaml` and
+   run `pub upgrade`.
+4. **Run `pub outdated`** to confirm that you've updated the dependencies.
+5. **Test** your app or package to confirm that it still works as expected.
+
+
+## Example
+
+You run `pub outdated` in the top directory of your package or app,
+in the same directory as your package or app's
+[`pubspec.yaml`](/tools/pub/pubspec) file.
+Here's an example of running `pub outdated`:
+
+```terminal
+$ pub outdated
+Package              Current   Upgradable  Resolvable  Latest    
+dependencies: all up-to-date
+
+dev_dependencies    
+build_web_compilers  1.2.3     1.2.3       2.10.0      2.10.0    
+
+transitive dependencies
+_fe_analyzer_shared  -         -           2.0.0       2.0.0     
+analyzer             0.35.4    0.35.4      0.39.5      0.39.5    
+...
+
+9  dependencies are constrained to versions that are older than a resolvable version.
+To update these dependencies, edit pubspec.yaml.
+```
+
+<span style="color:red">[PENDING: Is there a better example we should use?]</span>
+
+The preceding output shows that all regular dependencies are up to date,
+but one [dev dependency][] (`build_web_compilers`) is out of date.
+Several [transitive dependencies][] are also out of date.
+
+[dev dependency]: /tools/pub/dependencies#dev-dependencies
+[transitive dependencies]: /tools/pub/glossary#transitive-dependency
+
+
+To fix the dev dependency version,
+the easiest solution is to change the `build_web_compilers` entry
+in `pubspec.yaml` from `^1.0.0` to `^2.10.0`.
+For help on specifying versions, see the
+[version constraints documentation][constraints].
+
+[constraints]: /tools/pub/dependencies#version-constraints
+
+<span style="color:red">[PENDING: Improve the formatting of the following diff]</span>
+
+```diff
+-  build_web_compilers: ^1.0.0
++  build_web_compilers: ^2.10.0
+```
+
+{{site.alert.tip}}
+  To see what changed in the new version, you can look at the package's changelog,
+  which you can find in the package page on
+  [pub.dev.]({{site.pub}})
+  For an example, see the
+  [build_web_compilers changelog.]({{site.pub-pkg}}/build_web_compilers#-changelog-tab-)
```

Comment:
I think the link recently changed to: https://pub.dev/packages/build_web_compilers/changelog

Reply:
Oh, that's much a much nicer URL. Although clicking on the in-page link still takes me to the ugly URL.

@jonasfj can you confirm which URLs I should be using? Will the Installing tab URLs be updated soon?

Once I know what URLs to use, I should do a global search for package tab references.

Reply:
paging @jonasfj

Reply:
Yes, the URLs will change, but not in time for Dart 2.8, so I would suggest using the old URLs for now.

When the new URLs do launch I believe we will have javascript code redirecting the old ugly URLs such that these still work. We will want to update `dart.dev` docs when this happens to get a nicer experience. But it's not like `dart.dev` links are going to be broken all of a sudden.